Formula/gnupg.rb

```diff
@@ -38,12 +38,12 @@ class Gnupg < Formula
     depends_on "libidn"
   end
 
-  # Fixes a build failure without ldap.
+  # Fixes in-source-tree builds
   # Committed upstream, will be in the next release.
-  # https://dev.gnupg.org/T6239
+  # https://dev.gnupg.org/T6313
   patch do
-    url "https://dev.gnupg.org/rG7011286ce6e1fb56c2989fdafbd11b931c489faa?diff=1"
-    sha256 "407011d4ae9799f50008b431df60cd5b781dca0f572e956fd46245aa209af7e8"
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/6ce59559708c7f6acd025b5d11c536016c3129e5/gnupg/2.4.0-setup-scm.patch"
```

Comment:
Why not just from upstream?
```suggestion
    url "https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=patch;h=e89d57a2cb10bd04d266165015f159be2ab48984"
```

Reply:
I wasn't sure which approach was preferred. Have switched to that (will drop the formula-patch if this is merged).